wgpu-core/src/command/transfer.rs

```diff
@@ -136,6 +289,47 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         );
         barriers.extend(dst_pending.map(|pending| pending.into_hal(dst_buffer)));
 
+        assert_eq!(
+            size % BUFFER_COPY_OFFSET_ALIGNMENT,
+            0,
+            "Buffer copy size {} must be a multiple of {}",
+            size,
+            BUFFER_COPY_OFFSET_ALIGNMENT,
+        );
+        assert_eq!(
+            source_offset % BUFFER_COPY_OFFSET_ALIGNMENT,
+            0,
+            "Buffer source offset {} must be a multiple of {}",
+            source_offset,
+            BUFFER_COPY_OFFSET_ALIGNMENT,
+        );
+        assert_eq!(
+            destination_offset % BUFFER_COPY_OFFSET_ALIGNMENT,
+            0,
+            "Buffer destination offset {} must be a multiple of {}",
+            destination_offset,
+            BUFFER_COPY_OFFSET_ALIGNMENT,
+        );
+
+        let source_start_offset = source_offset;
+        let source_end_offset = source_offset + size;
+        let destination_start_offset = destination_offset;
+        let destination_end_offset = destination_offset + size;
+        assert!(
+            source_start_offset < src_buffer.size && source_end_offset <= src_buffer.size,
```

Comment:
can this be just a comparison for the `source_end_offset`? Looks like comparing the start offset is redundant

Reply:
Yeah, I was just favoring over-asserting vs under-asserting